### PR TITLE
Dynamic execution for browser

### PR DESCRIPTION
**Pull request by GokulBothe99**
Thank you for your contribution.
Before submitting this PR, please make sure:

- [ ] PR description and commit message should describe the changes done in this PR
- [ ] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [ ] Latest Code from master or specific release branch is merged to your branch
- [ ] No unwanted\commented\junk code is included
- [ ] No new warning upon build solution
- [ ] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve all Codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Update the Help Library document to match any feature changes

## Summary by CodeRabbit

- **New Features**
	- Enhanced command-line argument handling during application startup.
	- Introduced a method to split strings while preserving paths enclosed in quotes.
- **Bug Fixes**
	- Refined exception handling for clearer logging of startup errors.
	- Validated solution path before execution to prevent runtime issues.
- **Improvements**
	- Enhanced solution path handling with better formatting and validation logic.
	- Added robust parsing for command-line arguments, including checks for specific input patterns.

### CODE REVIEW

File: Ginger/Ginger/App.xaml.cs [hunk 1]
```diff
@@ -332,8 +341,30 @@ private bool ShouldStartGrid(string[] args)
         /// <returns>ParserResult containing parsed arguments or null.</returns>
         private ParserResult<object> ParseCommandLineArguments(string[] args)
         {
+            string[] arguments;
+            if (args.Length == 1 && System.Web.HttpUtility.UrlDecode(args[0]).Equals("ginger:///", StringComparison.OrdinalIgnoreCase))
```
Comment (Maheshkale447) [inline Ginger/Ginger/App.xaml.cs:345]: it should be "ginger://" instead of "ginger:///"

File: Ginger/Ginger/App.xaml.cs [hunk 2]
```diff
@@ -332,8 +341,30 @@ private bool ShouldStartGrid(string[] args)
         /// <returns>ParserResult containing parsed arguments or null.</returns>
         private ParserResult<object> ParseCommandLineArguments(string[] args)
         {
```
Comment (Maheshkale447) [inline Ginger/Ginger/App.xaml.cs:343]: Update the documentation for the function and also add documentation for each of the condition inside the function

File: Ginger/Ginger/App.xaml.cs [hunk 3]
```diff
@@ -332,8 +341,30 @@ private bool ShouldStartGrid(string[] args)
...
+                    input = input.Substring("ginger://".Length);
+                }
+                List<string> resultList = SplitWithPaths(input).Select(s => s.Trim('\"')).ToList();
+                arguments = resultList.ToArray();
+            }
+            else
+            {
+                arguments = args;
+            }
+            EditCommandlineArguments(arguments);
```
Comment (Maheshkale447) [inline Ginger/Ginger/App.xaml.cs:364]: do we need EditCommandlineArguments after UrlDecode on 352?

File: Ginger/Ginger/App.xaml.cs [hunk 4]
```diff
@@ -360,6 +391,10 @@ private DoOptions ExtractDoOptions(ParserResult<object> parserResult)
         /// <returns>True if the application is in execution mode, otherwise false.</returns>
         private bool IsExecutionMode(string[] args, DoOptions doOptions)
         {
+            if (args.Length == 1 && System.Web.HttpUtility.UrlDecode(args[0]).Equals("ginger:///", StringComparison.OrdinalIgnoreCase))
```
Comment (Maheshkale447) [inline Ginger/Ginger/App.xaml.cs:394]: check should be starting with ginger:// instead of equals

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

File: Ginger/GingerCoreNET/RunLib/CLILib/DoOptions.cs [hunk 5]
```diff
@@ -35,8 +38,35 @@ public enum DoOperation
...
+                }
+
+                if (value.EndsWith("/"))
+                {
+                    value = value.TrimEnd('/');
+                }
+
+                if (value.Contains("Ginger.Solution.xml"))
+                {
+                    value = Path.GetDirectoryName(value)?.Trim() ?? string.Empty;
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/GingerCoreNET/RunLib/CLILib/DoOptions.cs:63]: done

File: Ginger/GingerCoreNET/RunLib/CLILib/DoOptions.cs [hunk 6]
```diff
@@ -17,6 +17,9 @@ limitations under the License.
 #endregion
 
 using CommandLine;
+using System;
+using System.IO;
+using System.Web;
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/GingerCoreNET/RunLib/CLILib/DoOptions.cs:22]: done

File: Ginger/Ginger/App.xaml.cs (hunk 1 above)
Comment (GokulBothe99) [reply Ginger/Ginger/App.xaml.cs:345]: when user want to just open the Ginger from browser at that time one "/" slash is gets added so to handle that situation we have used this condition

File: Ginger/Ginger/App.xaml.cs (hunk 2 above)
Comment (GokulBothe99) [reply Ginger/Ginger/App.xaml.cs:343]: done

File: Ginger/Ginger/App.xaml.cs (hunk 3 above)
Comment (GokulBothe99) [reply Ginger/Ginger/App.xaml.cs:364]: no,removed

File: Ginger/GingerCoreNET/RunLib/CLILib/DoOptions.cs [hunk 7]
```diff
@@ -35,8 +38,35 @@ public enum DoOperation
...
+
+                if (lowerValue.Contains("ginger://"))
+                {
+                    value = WebUtility.UrlDecode(value.Replace("ginger://", "", StringComparison.OrdinalIgnoreCase));
+                }
+
+                if (value.EndsWith("/"))
+                {
+                    value = value.TrimEnd('/');
+                }
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/GingerCoreNET/RunLib/CLILib/DoOptions.cs:59]: it's  a more readable format

File: Ginger/GingerCoreNET/RunLib/CLILib/DoOptions.cs [hunk 8]
```diff
@@ -35,8 +38,35 @@ public enum DoOperation
...
+
         [Option('s', "solution", Required = true, HelpText = "Set solution folder")]
-        public string Solution { get; set; }
+        public string Solution
+        {
+            get => _solution;
+            set
+            {
+                string lowerValue = value.ToLowerInvariant();
+
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/GingerCoreNET/RunLib/CLILib/DoOptions.cs:50]: done

File: Ginger/Ginger/App.xaml.cs [hunk 9]
```diff
@@ -280,10 +283,36 @@ private async void Application_Startup(object sender, StartupEventArgs e)
...
+            var matches = Regex.Matches(input, pattern);
+            var results = new List<string>();
+
+            foreach (Match match in matches)
+            {
+                results.Add(match.Value.Trim());
+            }
 
+            return results;
+        }
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/Ginger/App.xaml.cs:303]: it's working

File: Ginger/Ginger/App.xaml.cs [hunk 10]
```diff
@@ -341,6 +391,10 @@ private DoOptions ExtractDoOptions(ParserResult<object> parserResult)
         /// <returns>True if the application is in execution mode, otherwise false.</returns>
         private bool IsExecutionMode(string[] args, DoOptions doOptions)
         {
+            if (args.Length == 1 && System.Web.HttpUtility.UrlDecode(args[0]).Equals("ginger:///", StringComparison.OrdinalIgnoreCase))
+            {
+                return false;
+            }
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/Ginger/App.xaml.cs:397]: handled

File: Ginger/GingerCoreNET/RunLib/CLILib/DoOptions.cs [hunk 11]
```diff
@@ -35,8 +38,35 @@ public enum DoOperation
...
+                if (lowerValue.Contains("ginger://"))
+                {
+                    value = WebUtility.UrlDecode(value.Replace("ginger://", "", StringComparison.OrdinalIgnoreCase));
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/GingerCoreNET/RunLib/CLILib/DoOptions.cs]: done

File: Ginger/Ginger/App.xaml.cs [hunk 12]
```diff
@@ -385,7 +415,7 @@ private void ProcessGingerUIStartup(DoOptions doOptions)
 
                 StartGingerUI();
 
-                if (doOptions != null && !string.IsNullOrWhiteSpace(doOptions.Solution))
+                if (doOptions != null && !string.IsNullOrWhiteSpace(doOptions.Solution) && Directory.Exists(doOptions.Solution))
                 {
                     DoOptionsHandler.Run(doOptions);
```
[the remark this answers is not in the document]
Comment (GokulBothe99) [reply Ginger/Ginger/App.xaml.cs:420]: done

**Review by Maheshkale447 — APPROVED** [review]

**Pull request merged**